### PR TITLE
AB#164

### PR DESCRIPTION
**Pull request by paola-quintanilla-jalafund**
[AB#164](https://dev.azure.com/funjaladev/7867a1c5-5ac7-4c7f-baff-87d740bf4fc1/_workitems/edit/164)

# Overview

- Integrate resumelist api with getResumes from resumes service 
- Add TDD
- Use api in component resume list

### CODE REVIEW

**Review by niver-copa-jalafund — DISMISSED** [review]
I don´t find any problems, for me.

**Review by DaleneyRuben — CHANGES_REQUESTED** [review]
@paola-quintanilla-jalafund Please fulfill the pr template, linking your azure story and adding a description for this PR.

**Review by freddy-perez-jalafund — DISMISSED** [review]

**Review by freddy-perez-jalafund — DISMISSED** [review]

**Review by niver-copa-jalafund — DISMISSED** [review]
Looks good,

File: src/components/resumes/ResumesAPI.js [hunk 1]
```diff
@@ -20,14 +24,15 @@ export async function postResume(resume) {
   }
 }
 
-export async function getResumes() {
+export default async function getResumes() {
```
Comment (DaleneyRuben) [inline src/components/resumes/ResumesAPI.js]: why are you changing this to default?

File: src/components/resumes/__tests__/ResumeList.test.js [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+const API_NAME = "resumes";
+const API_VERSION = "v1";
+const url = `${API_HOST}/api/${API_VERSION}/${API_NAME}`;
+
+test("should return a resume list", async () => {
+  axios.get.mockImplementationOnce(() => Promise.resolve(mockData));
+
+  await expect(getResumes()).resolves.toEqual(mockData.data);
+
+  expect(axios.get).toHaveBeenCalledWith(`${url}`);
```
Comment (DaleneyRuben) [inline src/components/resumes/__tests__/ResumeList.test.js]: you are testing axios here. your test name says "should return a resume list" you should test the component then

File: src/components/resumes/__tests__/ResumeList.test.js [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+test("should return false", async () => {
+  const resolve = true;
+
+  axios.get.mockImplementationOnce(() => Promise.resolve(new Error()));
+
+  expect(getResumes())
+    .rejects.toThrow(resolve)
+    .then()
+    .catch((err) => {});
+});
```
Comment (DaleneyRuben) [inline src/components/resumes/__tests__/ResumeList.test.js]: you don't need to test axios. remove this test

File: src/components/resumes/__tests__/ResumeList.test.js [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+        <Resumes />
+      </Route>
+    </MemoryRouter>
+  );
+  const resumesScreen = await screen.findByText(
+    "rodrigo.baldivieso@fundacion-jala.org"
+  );
+  expect(resumesScreen).toBeInTheDocument();
+  expect(axios.get).toHaveBeenCalledWith(url);
+});
```
Comment (DaleneyRuben) [inline src/components/resumes/__tests__/ResumeList.test.js]: 🤔 I saw this before maybe you wanna check again that [test](https://github.com/fundev3/teamup-webapp/blob/develop/src/components/projects/__tests__/Details.test.js)

**Review by DaleneyRuben — CHANGES_REQUESTED** [review]
Please merge your branch with develop

**Review by niver-copa-jalafund — DISMISSED** [review]
I was review your code, and I try, and is working

**Review by freddy-perez-jalafund — DISMISSED** [review]
Looks good for me

File: src/components/resumes/__tests__/ResumeList.test.js [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+const API_HOST = process.env.REACT_APP_API_RESUMES_URL;
+const API_NAME = "resumes";
+const API_VERSION = "v1";
+const url = `${API_HOST}/api/${API_VERSION}/${API_NAME}`;
+
+test("should return an empty list", async () => {
+  axios.get.mockImplementationOnce(() => Promise.resolve({ data: [] }));
+
+  await expect(getResumes()).resolves.toEqual([]);
+});
```
Comment (DaleneyRuben) [inline src/components/resumes/__tests__/ResumeList.test.js]: remove this we don't neeed to test axios

**Review by DaleneyRuben — CHANGES_REQUESTED** [review]
Just a small thing, please let me know when you're done and I can give my approval

File: src/components/resumes/__tests__/ResumeList.test.js (hunk 5 above)
Comment (paola-quintanilla-jalafund) [reply src/components/resumes/__tests__/ResumeList.test.js]: @DaleneyRuben done :D

**Review by DaleneyRuben — APPROVED** [review]
🚀

**Review by niver-copa-jalafund — APPROVED** [review]

**Review by freddy-perez-jalafund — APPROVED** [review]

**Pull request merged**